src/cpp/fastdds/publisher/DataWriterImpl.hpp

```diff
@@ -116,6 +116,33 @@ class DataWriterImpl
             void* data,
             const fastrtps::rtps::InstanceHandle_t& handle);
 
+    /*!
+     * @brief Implementation of the DDS `register_instance` operation.
+     * It deduces the instance's key and tries to get resources in the PublisherHistory.
+     * @param[in] instance Sample used to get the instance's key.
+     * @return Handle containing the instance's key.
+     * This handle could be used in successive `write` or `dispose` operations.
+     * In case of error, HANDLE_NIL will be returned.
+     */
+    fastrtps::rtps::InstanceHandle_t register_instance(
+            void* instance);
+
+    /*!
+     * @brief Implementation of the DDS `register_instance` and `dispose` operations.
+     * It sends a CacheChange_t  with the king to NOT_ALIVE_UNREGISTERED or NOT_ALIVE_DISPOSED,
```

Comment:
If we make the change I proposed above, change to `NOT_ALIVE_DISPOSED_UNREGISTERED`
```suggestion
     * It sends a CacheChange_t  with the king to NOT_ALIVE_UNREGISTERED or NOT_ALIVE_DISPOSED_UNREGISTERED,
```

Reply:
Updated documentation.